incubator/hnc/scripts/performance/README.md

```diff
@@ -0,0 +1,59 @@
+## Before you start
+
+This performance test is to test HNC's startup performance, including controller
+working time and number of object reads and writes during HNC startup and restart.
+
+Please be aware that the default performance test will create about maximum 500
+namespaces and 1000 objects in your cluster. To change the scale, you can edit
+`load-topology-` files to update the namespace numbers by `N` and object numbers
+by `O`.
+
+There are three topologies we support in this test to simulate possible use cases:
+```
+Wide - 1 root and N children (2-level-hierarchy)
+Full - 1 root, N children and N*N grandchildren (3-level-hierarchy)
+Skewer - 1 root, 1 child, 1 grandchild, 1 great grandchild, ... for N generations (N-level-hierarchy)
+```
+
+## Run performance test
+
+First, you will need to set the `enable-test-log` flag so that HNC can generate
+logs for this test to catch.
+
+To do so, add `- --enable-test-log` in `config/manager/manager.yaml` so
+container `args` will look like this:
+```
+args:
+- "--webhook-server-port=9443"
+# If /config/default/manager_auth_proxy_patch.yaml is used, the port in
+# --metrics-addr must match the value of --upstream passed to
+# kube-rbac-proxy.
+- "--metrics-addr=:8080"
+- "--max-reconciles=10"
+- "--apiserver-qps-throttle=50"
+- "--enable-internal-cert-management"
+- "--cert-restart-on-secret-refresh"
+- "--enable-test-log"        ####### <<<<---- Add this line
+```
+
+Run `make manifests` to update the manifest to include the flag.
+
+Now you can run performance test under the HNC root directory by:
+```
+.../incubator/hnc$. scripts/performance/test.sh
```

Comment:
I'd just start with the dollar sign, because everyone will have their shell configured a different way:

```
$ scripts/performance/test.sh
```

Also - what's that dot?

Reply:
I would get `Permission denied` without prefixing the script with `. `.

Reply:
Have you made it executable? `chmod +x path/to/script`.

Reply:
If I had to guess, I assume you need to do `chomd +x scripts/performance/test.sh` to fix the no dot issue.

Reply:
Ah thank you both, no. Made all scripts executable now!